feeder/api/models/kronos.py

```diff
@@ -67,12 +67,12 @@ class NewDevice(BaseModel):
         'softwareVersion': '2.8.0'
     }
     """
-    name: str
-    type: str
-    uid: str
+    name: Optional[str]
+    type: Optional[str]
+    uid: Optional[str]
     gatewayHid: str
-    softwareName: str
-    softwareVersion: str
+    softwareName: Optional[str]
```

Comment:
I feel like we should set this from the corresponding gateway, versus not storing anything at all. That makes the logic to surface it in the API/UI more complex down the line.

Reply:
This goes for both `softwareName` and `softwareVersion`.

Reply:
That seems reasonable since in our case Gateway is pretty much equal to Device. Although it's not directly related to this change, right?

Reply:
This comment is pointless because your gateway doesn't report it's version either. Unfortunately, we do need to make this optional on both the gateway and the device. 😞